### ISSUE TITLE
VAST or VMAP? Learn the difference, use cases and understand what we should use

### ISSUE THREAD

**Issue by nicolaslevy** [montevideo-tech/vast-2-sgai#3]
We need to evaluate the differences between VAST (Video Ad Serving Template) and VMAP (Video Multiple Ad Playlist) to determine which is better suited for our project goals. This involves understanding their use cases, capabilities, and potential integration points within our current ideas. The goal is to make an informed decision that aligns with our needs and ensures optimal ad serving functionality.

**Comment by nicolaslevy** [comment]
I updated the [whiteboard](https://app.diagrams.net/#G1MPodWl1R3DhgWXG54HC3dLo7qD7aTzZT#%7B%22pageId%22%3A%22efapASfb81Vfz7Dzpj0K%22%7D) with VMAP concepts and some key diagrams to understand its purpose.

![image](https://github.com/user-attachments/assets/22b66262-cd6c-4ea4-82bd-4ae79094a490)


In the [IAB VMAP 1.0.1](https://www.iab.com/wp-content/uploads/2015/06/VMAP.pdf) specification, several sections indicate that VMAP is designed for scenarios where ad break timings are predetermined:

Section 1.3: How VMAP Works: "VMAP represents a playlist structure that wraps one or more ad responses. This structure defines the ad breaks within a video program, identifying details such as how many ad breaks, which ad types to display, and when each ad break should occur." 

Section 1.5: Live or Simulcast Content:"While VMAP can conceivably be used to insert advertising into live or simulcast streaming content, the VMAP guideline does not include any special provisions for use in this environment or make any recommendations as to how to apply VMAP to live streams."

I understand that in the scenario we want to build, we don’t necessarily know the ad breaks that will occur in a live stream, and our primary use case is to handle the interstitial that arrives with an ad pod.

That said, I believe that at least at the beginning, we shouldn’t worry about VMAP and should only work with VAST (or, at most, a VMAP with a single VAST).

**Comment by nicolaslevy** [comment]
In last week's call, we discussed that there could be a case for requesting VMAP in VOD scenarios, although we understand this might not apply to live streams.

We updated the discussion with the diagram created by David H., which illustrates a possible implementation.

Additionally, we learned that in HLS, we can use query parameters as templates. These will be helpful for sending user parameters to the SGAI service.

![image](https://github.com/user-attachments/assets/dd5a7833-82e1-4054-9657-8624071bd412)
